### PR TITLE
modify transaction creation to allow offline interaction

### PR DESCRIPTION
**Pull request by Kunde21**
Creating a transaction from an offline location, especially for
remittance purposes, was not being handled in the transaction service
definition.

Allows for depositing money from a physical location (atm/branch) and
sending money for physical receipt (remittance transaction).  Note
that the transaction should have a bank account as source or
destination to be valid.

### CODE REVIEW

File: v1/types/types.gunk [hunk 1]
```diff
@@ -252,6 +252,26 @@ type Location struct {
...
 }
 
+type OfflineUserInfo struct {
+	// UserID
+	UserID string `pb:"1" json:"user_id"`
+
+	// FirstName of the person
+	FirstName string `pb:"2" json:"first_name"`
+
+	// MiddleName `pb:"" json:""` or middle names (space separated) of the person
```
Comment (grikwong) [inline v1/types/types.gunk]: I think `pb:"" json:""` annotation here is a typo? Same goes to MiddleName, LastName, and MobileNo fields?

Comment (Kunde21) [reply v1/types/types.gunk]: Fixed.

**Review by grikwong — APPROVED** [review]

File: v1/types/types.gunk [hunk 2]
```diff
@@ -252,6 +252,26 @@ type Location struct {
...
+	// UserID
+	UserID string `pb:"1" json:"user_id"`
+
+	// FirstName of the person
+	FirstName string `pb:"2" json:"first_name"`
+
+	// MiddleName or middle names (space separated) of the person
+	MiddleName string `pb:"3" json:"middle_name"`
+
+	// LastName of person
```
Comment (mvdan) [inline v1/types/types.gunk]: If you say "or middle names", might as well say "or last names" here too. Don't Filipinos have two surnames too, due to the Spanish influence?

**Review by mvdan — APPROVED** [review]

File: v1/types/types.gunk (hunk 2 above)
Comment (Kunde21) [reply v1/types/types.gunk]: True.  Changed

**Pull request merged**